casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala

```diff
@@ -13,12 +11,17 @@ import scala.util.{Failure, Success, Try}
 
 object DeployRuntime {
 
-  def propose[F[_]: DeployService: Monad](): F[Unit] =
+  def propose[F[_]: Monad: Capture: DeployService](): F[Unit] =
     DeployService[F].createBlock().flatMap {
       case Some(block) =>
-        DeployService[F].addBlock(block)
+        println(s"Response: Successfully created block")
+        for {
+          response <- DeployService[F].addBlock(block)
+          _        <- Capture[F].capture { println(s"Response: ${response._2}") }
+        } yield ()
 
       case None =>
+        println(s"Response: Failure! Did not create block")
```

Comment:
Please put this under a `Capture` and remove the `().pure[F]`. It rather misleading to have pure unit returned simply because the intended side-effect have been placed out of scope of the `F`.